proto/rill/runtime/v1/api.proto

```diff
@@ -954,6 +954,7 @@ message CompleteRequest {
   string instance_id = 1 [(validate.rules).string = {pattern: "^[_\\-a-zA-Z0-9]+$"}];
   string conversation_id = 2; // If not provided, creates a new conversation
   string prompt = 3;
+  rill.ai.v1.CompletionMessageContext context = 4;
```

Comment:
This is a general comment that I'll put here. Since this is specific to the router/analyst agents, I don't think it really belongs in the `proto/rill/ai` and `runtime/ai/ai.go` files, which contain generic AI abstractions, but don't deal in Rill-specific features like explores.

If you think about this from the perspective that we'll soon add more agents (e.g. for developing a model, or for explaining an error), we probably need some kind of `oneof` type in the `CompleteRequest` and `RouterAgentArgs` for passing this context through. Do you want to take a stab at that? Otherwise feel free to just flatten the args here into those types for now (we anyway generally have flattened types in the proto messages).